tools/PI/DevHome.PI/Views/BarWindowHorizontal.xaml.cs

```diff
@@ -408,33 +402,56 @@ public void SetRegionsForTitleBar()
 
     private void SetDefaultPosition()
     {
+        // First set our default size before setting out position
+        SetDefaultWidthAndHeight();
+
         // If attached to an app it should show up on the monitor that the app is on
-        _monitorRect = GetMonitorRectForWindow(_viewModel.ApplicationHwnd ?? TryGetParentProcessHWND() ?? ThisHwnd);
-        var screenWidth = _monitorRect.right - _monitorRect.left;
+        // Be sure to grab the DPI for that monitor
+        var dpiScale = GetDpiScaleForWindow(_viewModel.ApplicationHwnd ?? TryGetParentProcessHWND() ?? ThisHwnd);
+
+        RECT monitorRect = GetMonitorRectForWindow(_viewModel.ApplicationHwnd ?? TryGetParentProcessHWND() ?? ThisHwnd);
+        var screenWidth = monitorRect.right - monitorRect.left;
         this.Move(
-            (int)((screenWidth - (Width * _dpiScale)) / 2) + _monitorRect.left,
-            (int)WindowPositionOffsetY + _monitorRect.top);
+            (int)((screenWidth - (Width * dpiScale)) / 2) + monitorRect.left,
+            (int)WindowPositionOffsetY + monitorRect.top);

```

Comment:
>WindowPositionOffsetY [](http://example.com/codeflow?start=17&length=21)

Do we need to make these Offsets DPI aware?

Reply:
Played with this for a bit... I don't think so. Since the Move APIs take straight-up screen coordinates (not scaled for DPI), we don't need to do general scaling here, and 30 pixels down on a 100% DPI monitor is the same distance as 30 pixels on a 175% DPI. I think your math here was good.